### PR TITLE
Add Helm chart OCI release to GH automation

### PR DESCRIPTION
**Pull request by inteon**
Adds logic for pushing the Helm chart to an OCI registry (and adding a non-v-prefixed tag)

See https://github.com/cert-manager/makefile-modules/pull/219 for more info.

### CODE REVIEW

File: make/00_mod.mk [hunk 1]
```diff
@@ -43,12 +43,9 @@ api_docs_package := $(repo_name)/pkg/apis/trust/v1alpha1
 api_docs_branch := main
 
 helm_chart_source_dir := deploy/charts/csi-driver-spiffe
-helm_chart_name := cert-manager-csi-driver-spiffe
+helm_chart_image_name := quay.io/jetstack/charts/cert-manager-csi-driver-spiffe
 helm_chart_version := $(VERSION)
 helm_labels_template_name := cert-manager-csi-driver-spiffe.labels
-helm_docs_use_helm_tool := 1
-helm_generate_schema := 1
-helm_verify_values := 1
```
Comment (SgtCoDFish) [inline make/00_mod.mk:51]: question: Why are these removed? They don't seem related to what we're changing - are they defaulted on and therefore redundant to set here?

Comment (inteon) [reply make/00_mod.mk:51]: Yes, this is cleanup, the variables have been removed here: https://github.com/cert-manager/makefile-modules/pull/218

**Review by wallrj — APPROVED** [review]
Thanks for updating the release docs. I can't see a test release, so I assume you'll test this after merging.

Looks the same as the other PRs in other repos.

/approve
/lgtm
